### PR TITLE
DOC: Showcase usage without patching in readme

### PR DESCRIPTION
**Pull request by david-cortes-intel**
## Description

This PR extends a bit the main readme of the repository to mention that it is also possible to use sklearnex without patching, and gives out example snippets to do so. It also updates a dead link to the patching section in docs along the way.

---

PR should start as a draft, then move to ready for review state after CI is passed and all applicable checkboxes are closed.
This approach ensures that reviewers don't spend extra time asking for regular requirements.

You can remove a checkbox as not applicable only if it doesn't relate to this PR in any way.
For example, PR with docs update doesn't require checkboxes for performance while PR with any change in actual code should have checkboxes and justify how this code change is expected to affect performance (or justification should be self-evident).

Checklist to comply with **before moving PR from draft**:

**PR completeness and readability**

- [x] I have reviewed my changes thoroughly before submitting this pull request.
- [x] I have updated the documentation to reflect the changes or created a separate PR with update and provided its number in the description, if necessary.
- [x] Git commit message contains an appropriate signed-off-by string _(see [CONTRIBUTING.md](https://github.com/uxlfoundation/scikit-learn-intelex/blob/main/CONTRIBUTING.md#pull-requests) for details)_.
- [x] I have added a respective label(s) to PR if I have a permission for that.
- [x] I have resolved any merge conflicts that might occur with the base branch.

**Testing**

- [x] I have run it locally and tested the changes extensively.
- [x] All CI jobs are green or I have provided justification why they aren't.

**Performance**

Not applicable.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,13 +61,23 @@ With Extension for Scikit-learn, you can:
...
     from sklearnex import patch_sklearn
     patch_sklearn()
 
     from sklearn.cluster import DBSCAN
 
+    X = np.array([[1., 2.], [2., 2.], [2., 3.],
+                  [8., 7.], [8., 8.], [25., 80.]], dtype=np.float32)
+    clustering = DBSCAN(eps=3, min_samples=2).fit(X)
+    ```
+  - Or without patching:
```
Comment (napetrov) [inline README.md:75]: I'm ok with this format or alternatively we can show one case with patching and another without , not necessary showing both patched and not patched variants

**Comment by david-cortes-intel** [comment]
I now realize we have a sphinx plugin for arranging things into tabs, so I thought of showing the different ways of using sklearnex in the online docs as tabs:
image

**Review by napetrov — APPROVED** [review]
